test/why.js

```diff
@@ -359,7 +359,7 @@ test('functions', function (t) {
 		var genFnSpaceStar = Function('return function *() {};')();
 		var genNoSpaces = Function('return function*(){};')();
 		st.equal(
-			'second argument is a Generator; first is not',
+			'toStringTag is not the same: [object Function] !== [object GeneratorFunction]',
```

Comment:
`Object#toString` only depends on `Symbol.toStringTag` in later engines that have it - i'd be fine detecting support for this symbol, and making the message more specific when it's present tho.